src/taipy/core/data/parquet.py

```diff
@@ -0,0 +1,166 @@
+# Copyright 2022 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+from datetime import datetime, timedelta
+from os.path import isfile
+from typing import Any, Dict, List, Optional, Set
+
+import modin.pandas as modin_pd
+import pandas as pd
+
+from taipy.config.common.scope import Scope
+
+from ..common._reload import _self_reload
+from ..common.alias import DataNodeId, JobId
+from ..exceptions.exceptions import InvalidExposedType, MissingRequiredProperty
+from .data_node import DataNode
+
+
+class ParquetDataNode(DataNode):
+    """Data Node stored as a Parquet file.
+
+    Attributes:
+        config_id (str): Identifier of the data node configuration. This string must be a valid
+            Python identifier.
+        scope (Scope^): The scope of this data node.
+        id (str): The unique identifier of this data node.
+        name (str): A user-readable name of this data node.
+        owner_id (str): The identifier of the owner (pipeline_id, scenario_id, cycle_id) or `None`.
+        parent_ids (Optional[Set[str]]): The identifiers of the parent tasks or `None`.
+        last_edit_date (datetime): The date and time of the last modification.
+        job_ids (List[str]): The ordered list of jobs that have written this data node.
+        cacheable (bool): True if this data node is cacheable. False otherwise.
+        validity_period (Optional[timedelta]): The validity period of a cacheable data node.
+            Implemented as a timedelta. If _validity_period_ is set to None, the data_node is
+            always up-to-date.
+        edit_in_progress (bool): True if a task computing the data node has been submitted
+            and not completed yet. False otherwise.
+        path (str): The path to the Parquet file.
+        properties (dict[str, Any]): A dictionary of additional properties. The _properties_
+            must have a _"default_path"_ or _"path"_ entry with the path of the Parquet file:
+
+            - _"default_path"_ `(str)`: The default path of the Parquet file.\n
+            - _"columns"_ `(Optional[List[str]])`: If not None, only these columns will be read from the file.
```

Comment:
Here, we document the required properties of a ParquetDataNode that the user must provide.
`columns` and `compression` are optional though.

Reply:
Oh, so the optional parameters should only be in the docstring for `DataNodeConfig._configure_parquet`? I included those optional parameters in the `ParquetDataNode` docstring as well because [CSVDataNode](https://github.com/Avaiga/taipy-core/blob/7381d79e0dbd5324aaefb4b5f2f4bc3666bd8054/src/taipy/core/data/csv.py#L52-L53) listed the `has_header` and `exposed_type` optional parameters there.

Reply:
Oh, yes, this is just a list of properties, and `"path"` is required. My bad, you are actually correct on this 😄
We need to rethink about `"columns"` and update this accordingly. Thank you.